src/data/nat/cast.lean

```diff
@@ -98,6 +98,23 @@ eq_sub_of_add_eq $ by rw [← cast_add, nat.sub_add_cancel h]
 | (n+1) := (cast_add _ _).trans $
 show ((m * n : ℕ) : α) + m = m * (n + 1), by rw [cast_mul n, left_distrib, mul_one]
 
+theorem cast_dvd {α : Type*} [field α] {m n : ℕ} (n_dvd : n ∣ m) (n_nonzero : (n:α) ≠ 0) : ((m / n : ℕ) : α) = m / n :=
+ begin
+   rcases n_dvd with ⟨k, rfl⟩,
+   have : n ≠ 0, {rintro rfl, simpa using n_nonzero},
+   rw nat.mul_div_cancel_left _ (nat.pos_iff_ne_zero.2 this),
+   rw [nat.cast_mul, mul_div_cancel_left _ n_nonzero],
+ end
+
+ @[simp, norm_cast]
```

Comment:
Will norm_cast ever be able to use this, because of of `n_dvd`? (I'm asking, I don't know myself whether it's smart enough to find stuff like this)

Reply:
It does work - by which I mean a proof using `exact_mod_cast` failed when I removed the attribute.